### ISSUE TITLE
[Outlook][Mac][16.74] Office.CustomProperties.saveAsync time out

### ISSUE THREAD

**Issue by milankonir** [OfficeDev/office-js#3436]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Mac
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: 16.74 (23061100)
* Operating System: macOS Ventura 13.4
* Browser (if using Office on the web): N/A
 
## Expected behavior

`Office.CustomProperties.saveAsync` successfully saves custom properties

## Current behavior

`Office.CustomProperties.saveAsync` gets stuck indefinitely and never executes passed in callback

## Steps to reproduce

1. Load Gist below
2. Click run
3. `console.log('saved ok');` nor `console.log('save error', result.error)` never triggers

## Link to live example(s)

1. https://gist.github.com/milankonir/1e021844c92469ed2fea0b5b43dd685f

# Provide additional details

Only fails on 16.74, older version seem to be okish

## Context

We are using custom properties during `ItemSend` processing and due to bug in this API, email sending gets stuck indefinitely until the 5 minute time out.
Also the side panel usage that relies on this API is heavily impaired.
This needs to be fixed ASAP!!!

## Useful logs

- [ ] Console errors
- [ ] Screenshots
- [ ] Test file (if only happens on a particular file)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by exextoc** [comment]
@milankonir We are not able to reproduce this issue. Can you please share diagnostic ID

Steps to generate Diagnostics ID:
```

- Re-launch Outlook 
- Repro the issue mentioned
- Click on Help->Collect Diagnostics
- Share the ID generated.

```

**Comment by pavelstudeny** [comment]
Based on telemetry coming from 173 users, about 20% of the Office.CustomProperties.saveAsync calls actually succeed (540 out of 2650). For a comparison, the success rate on Outlook 16.73 is 100%, measured during the time of 16.74 rollout.

This issue doesn't seem to be user-dependent. There are users where saveAsync never returns, few user that always succeed and most users experience the time out with occasional successes.

**Comment by milankonir** [comment]
Hi @exextoc, my Outlook for Mac updated to 16.74.1 and I can no longer reproduce with the simple `Office.CustomProperties` `load` / `save` Gist.

I can still repro with the following Gist which mimics what we do in our add-in. We are invoking `CustomProperties` and `internetHeaders` APIs in parallel so perhaps this is some race condition?

Gist: https://gist.github.com/milankonir/4deb1e50a91a782b866314cebc003a50
Repro video: https://cl.outreach.io/YEu4y8ZP

**Comment by milankonir** [comment]
@exextoc Another followup, this seems to be a race condition/deadlock between the two save functions.
If the order of calls to `Office.CustomProperties.saveAsync` and `Office.context.mailbox.item.internetHeaders.setAsync` is swapped, the `Office.context.mailbox.item.internetHeaders.setAsync` times out instead.

See following repro video: https://share.getcloudapp.com/P8uXyypO

**Comment by milankonir** [comment]
Diagnostic ID: **5EP9UJNS**

This is for use case when `Office.CustomProperties.saveAsync` hangs.

**Comment by exextoc** [comment]
@milankonir Thanks for reporting this issue. We can repro it now. It has been put on our backlog. We unfortunately have no timelines to share at this point.

Internal tracking id: Office: 8050433

**Comment by exextoc** [comment]
@milankonir We have fixed this issue. Please confirm if it is working for you or not.

**Comment by milankonir** [comment]
@exextoc I have updated to `16.74.2` and I can no longer reproduce the issue there. Thank you!
